### PR TITLE
Fix/slider

### PR DESCRIPTION
**Pull request by jamesducky**
Closes [[BUG] PlmgSlider - Component crash when causing re-render](https://app.asana.com/0/1198920871936085/1202735274432180/f)

### Summary of changes included in this PR

- Fix changes array to string. Component converts this string to an array.
- Added e2e tests.

### Reviewer checklist:

- Tests for
  - accessibility
  - rendered HTML (spec)
- Storybook stories for all variants
- JSDoc documentation for component
- Update Example app (React, ...)

### CODE REVIEW

File: packages/component-library/src/components/plmg-slider/test/plmg-slider.e2e.ts [hunk 1]
```diff
@@ -3,33 +3,149 @@ import { AxePuppeteer } from '@axe-core/puppeteer';
...
+    await page.setContent(
+      '<plmg-slider name="Range Slider" range-values="0, 10"></plmg-slider>'
+    );
+    const element = await page.find('plmg-slider >>> input[type="range"]');
+    expect(element).not.toBe(null);
+  });
+});
 
-    const htmlContent = `<plmg-slider name='Range Slider' range='0, 100'></plmg-slider>`;
+describe('sets the range and initial valuess', () => {
```
Comment (nathan-bird) [inline packages/component-library/src/components/plmg-slider/test/plmg-slider.e2e.ts:24]: I suggest adding "plmg-slider" to your test cases to be explicit they relate to this compeonent.
```suggestion
describe('plmg-slider sets the range and initial valuess', () => {
```

Comment (nathan-bird) [inline packages/component-library/src/components/plmg-slider/test/plmg-slider.e2e.ts:24]: Fix typo
```suggestion
describe('sets the range and initial values', () => {
```

File: packages/component-library/src/components/plmg-slider/test/plmg-slider.e2e.ts [hunk 2]
```diff
@@ -3,33 +3,149 @@ import { AxePuppeteer } from '@axe-core/puppeteer';
...
+    await page.setContent(
+      '<plmg-slider name="Range Slider" range-values="0.1, 9.99" default-value="-0.1"></plmg-slider>'
+    );
+    const element = await page.find('plmg-slider');
+    expect(element).toEqualAttribute('value', '0.1');
+  });
+});
 
-    await page.setContent('<main>' + htmlContent + '</main>');
+describe('displays marks', () => {
```
Comment (nathan-bird) [inline packages/component-library/src/components/plmg-slider/test/plmg-slider.e2e.ts:68]: I suggest adding "plmg-slider" to your test cases to be explicit they relate to this component.
```suggestion
describe('plmg-slider displays marks', () => {
```

File: packages/component-library/src/components/plmg-slider/test/plmg-slider.e2e.ts [hunk 3]
```diff
@@ -3,33 +3,149 @@ import { AxePuppeteer } from '@axe-core/puppeteer';
...
 
-    const results = await new AxePuppeteer(page as unknown as Page)
-      .disableRules([
-        'document-title',
-        'html-has-lang',
-        'landmark-one-main',
-        'page-has-heading-one',
-      ])
-      .analyze();
+describe('re-renders when props change', () => {
```
Comment (nathan-bird) [inline packages/component-library/src/components/plmg-slider/test/plmg-slider.e2e.ts:87]: I suggest adding "plmg-slider" to your test cases to be explicit they relate to this component.
```suggestion
describe('plmg-slider re-renders when props change', () => {
```

File: packages/component-library/src/components/plmg-slider/test/plmg-slider.e2e.ts [hunk 4]
```diff
@@ -3,33 +3,149 @@ import { AxePuppeteer } from '@axe-core/puppeteer';
...
+      },
+      props
+    );
+    await page.waitForChanges();
+    expect(await page.find('plmg-slider >>> .plmg.marks')).toBe(null);
+  });
+});
 
-    expect(results.violations).toHaveLength(0);
+describe('updates the value', () => {
```
Comment (nathan-bird) [inline packages/component-library/src/components/plmg-slider/test/plmg-slider.e2e.ts:128]: I suggest adding "plmg-slider" to your test cases to be explicit they relate to this component.
```suggestion
describe('plmg-slider updates the value', () => {
```

**Comment by jamesducky** [comment]
Changes made. 
Ignore the change log commit- it hasn't changed, but I uncommitted it by mistake, so had to recommit.

**Review by nathan-bird — APPROVED** [review]

**Pull request merged**